### PR TITLE
[SW-735] Update spot_ros2 for the new ArmImpedanceCommandStatus enum

### PR DESCRIPTION
**Pull request by khughes-bdai**
## Change Overview

Per the [4.0.0 release notes](https://dev.bostondynamics.com/docs/release_notes) there is a new enum in `ArmImpedanceFeedbackStatus` called `STATUS_TRAJECTORY_CANCELLED` (view details with `ros2 interface show bosdyn_api_msgs/msg/ArmImpedanceCommandFeedbackStatus`). If we receive a feedback message from a robot command with this status in `spot_ros2`, this will indicate that the Goal Response has failed. 

## Testing Done

- [x] All `spot_examples` successful (`walk_forward`, `arm_simple`, `send_inverse_kinematics_requests`)
- [x] internal examples using arm impedance control still work (`example_impedance_control.py`)
- [x] Manually try setting status to `STATUS_TRAJECTORY_CANCELLED` within `spot_ros2.py` to force a cancellation, verify that it gets cancelled on same internal arm impedance control examples. 
- [x] new unit tests passed

### CODE REVIEW

**Comment by khughes-bdai** [comment]
* **#301** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/301?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @khughes-bdai and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: spot_driver/spot_driver/spot_ros2.py [hunk 1]
```diff
@@ -1926,7 +1926,8 @@ def _process_synchronized_arm_command_feedback(self, feedback: ArmCommandFeedbac
                 return GoalResponse.FAILED
         elif choice == fb.FEEDBACK_ARM_IMPEDANCE_FEEDBACK_SET:
             if (
-                fb.arm_impedance_feedback.status.value == fb.arm_impedance_feedback.status.STATUS_TRAJECTORY_STALLED
+                fb.arm_impedance_feedback.status.value == fb.arm_impedance_feedback.status.STATUS_TRAJECTORY_CANCELLED
+                or fb.arm_impedance_feedback.status.value == fb.arm_impedance_feedback.status.STATUS_TRAJECTORY_STALLED
```
Comment (amessing-bdai) [inline spot_driver/spot_driver/spot_ros2.py:1930]: Is there a unit test for this?

**Review by bhung-bdai — APPROVED** [review]
Approved as long as you provide a test if possible

**Comment by amessing-bdai** [comment]
> Approved as long as you provide a test if possible

There is an existing test that should be updated.

**Review by amessing-bdai — APPROVED** [review]

**Comment by khughes-bdai** [comment]
### Merge activity

* **Mar 28, 4:04 PM EDT**: @khughes-bdai merged this pull request with [Graphite](https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/301).

**Pull request merged**